src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -653,6 +653,20 @@ void GcsServer::InitGcsWorkerManager() {
 }
 
 void GcsServer::InitGcsAutoscalerStateManager() {
+  RAY_CHECK(kv_manager_) << "kv_manager_ is not initialized.";
+  RAY_LOG(INFO) << "Autoscaler V2 enabled: "
+                << std::to_string(RayConfig::instance().enable_autoscaler_v2());
+  kv_manager_->GetInstance().Put(
+      kGcsAutoscalerStateNamespace,
+      kGcsAutoscalerV2EnabledKey,
+      std::to_string(RayConfig::instance().enable_autoscaler_v2()),
+      /*overwrite=*/true,
+      [](bool success) {
+        if (!success) {
+          RAY_LOG(WARNING) << "Failed to put autoscaler state to GCS.";
+        }
+      });
```

Comment:
Should we crash here?

Reply:
maybe we could eventually when we roll out the feature. I am a bit concerned this codepath gets triggered and cause behaviour change when this put failed before we flip the feature flag... 

So if it fails, it will behave as if autoscaler v2 not enabled to client.

Reply:
if this fails, it's default v1. From server's perspective, v1/v2 actually doesn't make a difference, it's just v2 will support a few more rpc calls. 

But sure, I am open to crashing here.